pkg/controller/sidecarset/sidecarset_controller.go

```diff
@@ -115,9 +112,8 @@ var _ reconcile.Reconciler = &ReconcileSidecarSet{}
 // ReconcileSidecarSet reconciles a SidecarSet object
 type ReconcileSidecarSet struct {
 	client.Client
-	scheme             *runtime.Scheme
-	updateExpectations expectations.UpdateExpectations
```

Comment:
Why did you remove this filed? My understanding of the previous bug had nothing to do with this field being defined here, right?

Reply:
Only updateExpectations were removed because they were initialized in utils

Reply:
A utils is added so that the sidecarset handler can access it, similar to how cloneset handles it

Reply:
you can move the function to file https://github.com/openkruise/kruise/blob/master/pkg/control/sidecarcontrol/util.go.

Reply:
OK, I've moved it